File: Selenium/Activities/Activity4.py
# Import required packages from selenium

from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager

# Set up the Firefox Driver using WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:

    # Launch the URL and get the title
    driver.get("https://v1.training-support.net/selenium/target-practice")
    home_title=driver.title
    print ("Title of the Home Page is : {}".format(home_title))

    # Find and get the header3 text
    header3=driver.find_element(By.XPATH, "//h3[@id='third-header']")
    header3_text= header3.text
    print ("Header3 text : " + header3_text)

    # Find and get the header5 color
    header5=driver.find_element(By.XPATH, "//h5[text()='Fifth header']")
    header5Color= header5.value_of_css_property("color")
    print("Header5 Color : {}".format(header5Color))

    # Find and get the Text of Violet Button
    violetButton=driver.find_element(By.XPATH, "//button[text()='Violet']")
    violetButtonClasses= violetButton.get_attribute("class")
    print("Violet Button Class : " + violetButtonClasses)

    # Find and get the Text of Grey Button
    greyButton=driver.find_element(By.XPATH, "//button[text()='Grey']");
    greyButtonColor= greyButton.text
    print("Grey Button Color : " + greyButtonColor)

    # CLose the Browser
    driver.close()

File: Selenium/Activities/Activity4.java
package activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;

public class Activity4 {
    public static void main(String[] args) {

        // Setup Firefox webdriver
        WebDriverManager.firefoxdriver().setup();

        // Instantiate the webdriver
        WebDriver driver = new FirefoxDriver();

        // Launch the Application URL and print the title of homepage
        driver.get("https://v1.training-support.net/selenium/target-practice");
        String homeTitle=driver.getTitle();
        System.out.println("Title of the Home Page is : " + homeTitle);

        // Find and get the header3 text
        WebElement header3=driver.findElement(By.xpath("//h3[@id='third-header']"));
        String header3Text= header3.getText();
        System.out.println("Header3 text : " + header3Text);

        // Find and get the header5 color
        WebElement header5=driver.findElement(By.xpath("//h5[text()='Fifth header']"));
        String header5Color= header5.getCssValue("color");
        System.out.println("Header5 Color : " + header5Color);

        // Find and get the Text of Violet Button
        WebElement violetButton=driver.findElement(By.xpath("//button[text()='Violet']"));
        String violetButtonClasses= violetButton.getAttribute("class");
        System.out.println("Violet Button Class : " + violetButtonClasses);

        // Find and get the Text of Grey Button
        WebElement greyButton=driver.findElement(By.xpath("//button[text()='Grey']"));
        String greyButtonColor= greyButton.getText();
        System.out.println("Grey Button Color : " + greyButtonColor);

        // CLose the Browser
        driver.close();
    }
}
